### DISCUSSION TITLE
如何在WPF中实现一个类似Web的基本权限管理框架

### DISCUSSION THREAD

**Discussion by luchong0813** [BYJRK/DotNet-Discussions#220] [WPF 开发]
我想实现类似Web中常见的权限管理功能，比如创建角色，角色赋予对应权限(用户管理->新增、删除、查询...)，为用户绑定对应角色，用户登录呈现对应角色菜单，我一说大家应该都懂，但实际操作起来，还是有点难下手，各位仁者见仁，智者见智，提供下思路，最好能附上源码或资料就更好了。
1、应用怎么知道自己有哪些模块，比如用户管理，用户管理又有CRUD等等，使用特性标记？通过反射获取对应特性从而生成Tree结构？
2、所有的模块菜单子级操作按钮，不打算存放在数据库，按照Web前端常见的做法，一般是由前端生成，通过接口把自己的菜单路径啥的发给后端存起来。在客户端中，自己能连数据库的情况下，怎么把某个角色对应的权限功能，存储在数据库中，以什么规则进行。（这里说的有点啰嗦，简单点说就是如何优雅的让角色与模块对应上）
3、角色对应的权限有了，应用自己有哪些模块(权限)也能生成了，登陆完，拿到该用户的角色后，怎么优雅的构建出来？

**Comment by xingzuhui** [comment]
你这个权限就是做起来,就是简陋的.
你说的这些都是功能权限的部分.
我知道的,还是数据权限,字段权限.
权限又分能不能看到,看到能不能使用,能使用能不能增删改.

说起来是不是抽象了,比如一hr系统,   包括人员基本信息和奖惩\工作经历等.
那么我的权限系统有可能是这样安排的,
1.普通员工只能看自己部门的普通员工的一般信息,不能看联系方式和别的部门的信息.
2.主管可以查看自己的员工信息,奖池信息,联系方式
3.经理可以查看修改自己部门的所有信息,查看别的部门的一般信息
4.集团hr可以修改各个公司的所有信息.
5.集团老板只能操作,不能新增,可以添加奖惩.

这个角色和人员的权限都是可以定义的.
